engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -287,26 +287,23 @@ def scc_activate_product
           auth = nil
           auth = request.headers['HTTP_AUTHORIZATION'] if request.headers.include?('HTTP_AUTHORIZATION')
           mode = nil
+          token_not_provided = !params.key?(:token) || params[:token].empty?
           if @system.byos?
             mode = 'byos'
           elsif !@product.free? && @product.extension?
-            regcode_error_message = 'A registration code is required to activate this product'
-            raise ActionController::TranslatedError.new(regcode_error_message) if (@system.payg? || @system.hybrid?) && params[:token].nil?
-
-            # system is not BYOS and
-            # product is a non free extension
-            # and a registration code is provided
-            mode = 'hybrid'
-            # the extensions must be the same version and arch
-            # than base product
-            base_prod = @system.products.find_by(product_type: :base)
-            if base_prod.present? && @product.arch == base_prod.arch && @product.version == base_prod.version
-              request.headers['proxy_byos_mode'] = mode
-              request.headers['scc_login'] = @system.login
-              request.headers['scc_password'] = @system.password
-              request.headers['instance_data'] = params[:instance_data]
-              request.headers['hwinfo'] = params[:hwinfo]
-              response = SccProxy.announce_system_scc(auth, request.headers)
+            unless token_not_provided
+              mode = 'hybrid'
+              # the extensions must be the same version and arch
+              # than base product
+              base_prod = @system.products.find_by(product_type: :base)
+              if @system.payg? && base_prod.present? && @product.arch == base_prod.arch && @product.version == base_prod.version
```

Comment:
announce the `hybrid` system to SCC just the first time

Reply:
So the condition to announce the system to SCC is `@system.payg? && base_prod.present?`?

Reply:
if it is the fake announcement but yes, in the subsequent PR to de-register/de-activate I handle the removal of this announcement if activation failed, as expected